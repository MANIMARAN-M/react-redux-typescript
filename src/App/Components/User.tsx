import React from "react";
import ListUsers from "./ListUsers";

interface UserDetails {
  name: string;
  data: any;
}

class User extends React.Component<{}, UserDetails> {
  constructor(props: any) {
    super(props);
    this.state = {
      name: "hey",
      data: [],
    };
  }
  componentDidMount?(): void {
    fetch("https://jsonplaceholder.typicode.com/users")
      .then((res) => res.json())
      .then((res) => {
        this.setState({ ...this.state, data: res });
      })
      .catch((err) => console.log(err));
  }
  render(): JSX.Element {
    console.log("render 3");
    return (
      <div data-testid="users-data">
        <div className="UsersListTitle">
          <h1>Fetching users data using TypeScript</h1>
        </div>
        <div>
          <ListUsers UserData={this.state.data} />
        </div>
      </div>
    );
  }
}

export default React.memo(User);
