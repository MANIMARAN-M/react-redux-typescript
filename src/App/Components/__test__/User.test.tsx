import User from "../User";
import { shallow, configure } from "enzyme";
import "@testing-library/jest-dom/extend-expect";
import { cleanup, screen, render } from "@testing-library/react";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
afterEach(cleanup);
describe("User component checking", () => {
  it("Check id in the component", () => {
    render(<User />);
    const userData = screen.getByTestId("users-data");
    expect(userData).toBeInTheDocument();
  });
  var component: any;
  component = shallow(<User />);
  test("It should mount", () => {
    expect(component.length).toBe(1);
  });
  it("should render our Snapshots correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
