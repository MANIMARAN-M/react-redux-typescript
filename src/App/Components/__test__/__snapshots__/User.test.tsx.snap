// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component checking should render our Snapshots correctly 1`] = `ShallowWrapper {}`;
